### ISSUE TITLE
Errors on presumed MDX expression with curly braces

### ISSUE THREAD

**Issue by dave-db** [mdx-editor/editor#318]
A string in `{curly braces}` within the body of a document is being parsed as an MDX expression, then throwing as there's no node handler for it. The same text content is processed ok by MDX evaluate/parse itself, so it looks like it relates to the conversion pipeline between markdown and Lexical.

## Example content

```md
Find your video at:

https://player.evideo.com/video?video-id={your-video-id}

> Replace `your-video-id` with your video id
```

The part in `{curly braces}` blew up the editor, but not the MDX processor (mdx-js evaluateSync), which rendered it fine unescaped. The editor error was:

```
Error: Cannot handle unknown node `mdxTextExpression`
    at Object.unknown (webpack-internal:///(:3000/admin/app-pages-browser)/./node_modules/@mdxeditor/editor/node_modules/mdast-util-to-markdown/lib/index.js:120:11)
    at Object.one [as handle] (webpack-internal:///(:3000/admin/app-pages-browser)/./node_modules/zwitch/index.js:112:17)
    at toMarkdown (webpack-internal:///(:3000/admin/app-pages-browser)/./node_modules/@mdxeditor/editor/node_modules/mdast-util-to-markdown/lib/index.js:85:24)
    at visit (webpack-internal:///(:3000/admin/app-pages-browser)/./node_modules/@mdxeditor/editor/dist/importMarkdownToLexical.js:78:156)
    at eval (webpack-internal:///(:3000/admin/app-pages-browser)/./node_modules/@mdxeditor/editor/dist/importMarkdownToLexical.js:68:45)
```

_Originally posted by @dave-db in https://github.com/mdx-editor/editor/discussions/312#discussioncomment-8171127_

**Comment by petyosi** [comment]
Thank you, I will test that. To ensure I'm not missing something, in that case, you would rather not have any special processing of the `{your-video-id}` token, right?

**Comment by dave-db** [comment]
Yup, absolutely - it should be just string content in the link (in this case). Looking at the docs for [mdast-util-mdx-expression](https://github.com/syntax-tree/mdast-util-mdx-expression?tab=readme-ov-file#use), I think the whitespace around braces should be treated significant, although I'm not clear on exactly what the spec is. `mdx-js` itself doesn't try to parse it (at least in my pipeline), so maybe I don't have that plugin enabled for rendering pages?

**Comment by doxman** [comment]
Hi, I'm having a possibly related problem with markdown containing curly braces. Thought I would check here before opening a new issue.

The scenario is, if my editor is using `jsxPlugin`, and my markdown contains non-escaped curly braces **wrapping text with at least one space**, then the editor breaks. So, `{foo bar}` causes breakage, but `{foo}` or `\{foo bar}` does not.

For example, I modified `src/examples/insert-markdown.tsx` so that `jsxMarkdownContentToInsert` has a line containing `{foo bar}` at the end; here's a screenshot of the error:

<img width="601" height="282" alt="Image" src="https://github.com/user-attachments/assets/f440e3d2-9031-409b-850e-874081322ec5" />

I suspect the bug comes from one `mdx-editor`'s dependencies - possibly `micromark` or `mdast-util`. But unfortunately, there is no error trace in the browser console. In fact, the "Get Markdown" button from the example still works, and logs the whole body including the `{foo bar}` portion.

Did either of you see anything like this while investigating the previous issue?

**Comment by petyosi** [comment]
Apologies for the slow follow-up here — this has been resolved since v2.17.0 (via #396), and I should have closed the loop on this issue earlier.

MDX expressions like `{your-video-id}` are handled by the `jsxPlugin`, which needs to be included in your plugins array:

```tsx
<MDXEditor
  markdown={markdown}
  plugins={[jsxPlugin()]}
/>
```

I verified that the exact markdown from this report renders correctly with the plugin enabled, including the URL containing `{curly braces}`.

Note that without the plugin, the parser still treats `{...}` as an MDX expression (the MDX syntax extensions are always active), but the failure is now caught and surfaced through the [`onError`](https://mdxeditor.dev/editor-api/interface/mdxeditorprops#_gutter_onError) prop rather than crashing the editor.

If I've misunderstood your use case and the plugin doesn't cover it, please let me know and I'll reopen.
